### PR TITLE
add automatic tick format detection

### PR DESCRIPTION
**Pull request by gka**
this is the basis for our upcoming new automatic axis tick formatting feature. this could as well live in the axis block, but for now I will just keep it in `shared`.

### CODE REVIEW

File: autoTickFormat.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+ * autoTickFormatDate([new Date(2000,0,1), new Date(2002,0,1)], 'quarter'); // 'YYYY|[Q]Q'
+ *
+ * @param {array} range - [min, max] of the data
+ * @param {string} precision - the input data precision (year|quarter|month|week|day|...)
+ * @returns {string} - day.js compatible format string
+ */
+export function autoTickFormatDate(range, precision = 'day') {
+    const span = Math.abs(range[1].getTime() - range[0].getTime());
+    if (precision === 'year')
+        return span > YEAR * 5 ? '´YY' : span > YEAR * 3 ? 'YYYY' : 'YYYY|MMM';
```
Comment (sto3psl) [inline autoTickFormat.js]:
```suggestion
        return span > YEAR * 5 ? 'YY' : span > YEAR * 3 ? 'YYYY' : 'YYYY|MMM';
```

Looks like a typo to me or does this mean anything in moment/dayjs? Couldn't find it in the dayjs API reference.

File: autoTickFormat.test.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
+import test from 'ava';
+import { autoTickFormatDate, autoTickFormatNumber } from './autoTickFormat';
```
Comment (sto3psl) [inline autoTickFormat.test.js]: Do these functions need to take care of "bad" data like `NaN` or `null`?

File: autoTickFormat.js (hunk 1 above)
Comment (gka) [reply autoTickFormat.js]: it's not a typo, that's for abbreviated years that we don't want to look like `19` but `´19`

Comment (gka) [reply autoTickFormat.js]: hm maybe the correct punctuation would be the simple apostrophe, actually.

File: autoTickFormat.test.js (hunk 2 above)
Comment (gka) [reply autoTickFormat.test.js]: no they only return a format string to be used by dayjs and numeral. and those are the ones to take care of invalid values. the only data values that go into `autoTickFormat` is the domain, which is expected to be valid numbers or dates.

**Pull request merged**